### PR TITLE
Implement Stained glass and stained glass pane

### PR DESCRIPTION
**Pull request by daniktheboss**
## Introduction
Adds stained glass and stained glass pane

## Tests
Tested, has all stained glass

### CODE REVIEW

**Comment by xZeroMCPE** [comment]
tf

**Comment by daniktheboss** [comment]
@xZeroMCPE ??

**Comment by TheDiamondYT1** [comment]
Why not just construct a GlassBlock/Pane and set the name? Would eliminate the need for a custom class

**Comment by TheDiamondYT1** [comment]
Also this is 1.2 change, so commit to the 1.2 branch

**Comment by mal0ne-23** [comment]
@TheDiamondYT1 Doesn't stained glass have a completely different ID than normal glass though? Correct me if I'm wrong

**Comment by TheDiamondYT1** [comment]
@mal0ne-23 https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/block/Block.php#L418

The first parameter is the id

**Comment by mal0ne-23** [comment]
@xZeroMCPE @AppleDevelops Also, I understand you guys don't like Danik, but keep that out of here. It wasn't copy pasted/stolen from anywhere and doesn't have any errors.

**Comment by xZeroMCPE** [comment]
@daniktheboss Who says' I don't like him? You should definitely verify where you get your information; before stating it. 

I'd also recommended you to keep all this out of this issue.

**Comment by TheDiamondYT1** [comment]
@xZeroMCPE You shouldnt have responsed in the first place

**Comment by dktapps** [comment]
@TheDiamondYT1 no. If you look at the majority of them they have a constructor override which only accepts meta. Regardless, that wouldn't make sense in this case because stained glass is a coloured block and each of its _variants_ have different meta.

This pull request looks reasonable, however it would be preferable if `StainedGlass` extended `Glass` and `StainedGlassPane` extended `GlassPane`, since they share all the same properties except ID and names.

**Comment by dktapps** [comment]
This can be accepted on master, stained glass already exists in the game since 1.0, it just wasn't properly implemented until 1.2.

File: src/pocketmine/block/StainedGlass.php [hunk 1]
```diff
@@ -26,7 +26,7 @@
 use pocketmine\block\utils\ColorBlockMetaHelper;
 use pocketmine\item\Item;
 
-class StainedGlass extends Transparent{
+class StainedGlass extends StainedGlass{
```
Comment (TheDiamondYT1) [inline src/pocketmine/block/StainedGlass.php]: Ummmmm

**Comment by TheDiamondYT1** [comment]
Oh whoops

**Comment by QuanFromChina** [comment]
@TheDiamondYT1 Stop spam your stupid " whoops comment " ! 130 kid watch this project ! Shut up !

**Comment by TheDiamondYT1** [comment]
@QuanFromChina Says you https://github.com/pmmp/PocketMine-MP/issues/1211

**Pull request merged**